docs/proposal/004-gateway-api-annotation-placement.md

```diff
@@ -0,0 +1,539 @@
+
+### Current Implementation Behavior
+
+Based on the source code
+([source/gateway.go](https://github.com/kubernetes-sigs/external-dns/blob/master/source/gateway.go)):
+
+**Gateway resource annotations:**
+
+- `external-dns.alpha.kubernetes.io/target` - read from Gateway
+  ([line ~380](https://github.com/kubernetes-sigs/external-dns/blob/master/source/gateway.go#L380))
+
+**Route resource annotations (HTTPRoute, GRPCRoute, TLSRoute, TCPRoute, UDPRoute):**
+
+- `external-dns.alpha.kubernetes.io/hostname` - read from Route
+- `external-dns.alpha.kubernetes.io/ttl` - read from Route
+- `external-dns.alpha.kubernetes.io/controller` - read from Route
+- **Provider-specific annotations** (e.g., `cloudflare-proxied`, `aws/*`, `scw/*`, etc.) - read from Route
+  ([line ~242](https://github.com/kubernetes-sigs/external-dns/blob/master/source/gateway.go#L242))
+
+This separation aligns with Gateway API architecture:
+
+- **Gateway** = infrastructure layer (IP addresses, listeners, load balancers)
+- **Routes** = application layer (DNS records, routing rules, hostnames)
+
+However, users expect provider-specific annotations to work on Gateway (similar to how `target` works), leading to silent failures.
+
+### Goals
+
+- Clarify annotation placement in documentation to prevent user confusion
+- Provide practical examples for common providers (Cloudflare, AWS, Scaleway)
+- Define a clear, documented contract for where each annotation type should be placed
+- Reduce support burden from repeated misconfigurations
+
+### Non-Goals
+
+- This proposal does not address the broader annotation standardization effort discussed in [PR #5080](https://github.com/kubernetes-sigs/external-dns/pull/5080)
+- Redesigning the Gateway API source implementation
+- Changing behavior for non-Gateway sources (Ingress, Service, etc.)
+- Making breaking changes to existing Gateway API functionality
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Platform Engineer with Cloudflare (#5901)
+
+*As a platform engineer*, I set up a Gateway with the `external-dns.alpha.kubernetes.io/cloudflare-proxied: "true"`
+annotation, expecting all DNS records for Routes using this Gateway to be proxied through Cloudflare. However, the
+annotation is silently ignored, and records are created without proxy, leading to unexpected traffic routing and
+security issues.
+
+**Root cause**: Had to dive into source code to discover that provider-specific annotations are only read from
+Route resources, not Gateway resources.
+
+**Current workaround**: Must copy the `cloudflare-proxied` annotation to every HTTPRoute manually.
+
+#### Story 2: User Attempting Route-Specific Targets (#4056)
+
+*As a user*, I want to specify different target DNS records for specific hosts while sharing a common Gateway. I
+added `external-dns.alpha.kubernetes.io/target` annotation on HTTPRoute to override the Gateway's target for one
```

Comment:
🤔 Would [this PR](https://github.com/kubernetes-sigs/external-dns/pull/5959) solve this use case ?